tmpl/blog/functoria.md

```diff
@@ -0,0 +1,153 @@
+
+For the last two month, I have been at OCamllabs for « holidays » with the grand task
+of « fixing the mirage tool ».
+
+I'm happy to present [Functoria](https://github.com/Drup/Functoria), a library to create arbitrary mirage-like DSLs. Functoria is independent from Mirage and will replace all the core engine that was bolted on the mirage tool until now.
+
+The bad news is that it's going to break some (little) things. To see the breaking changes and how to fix them, consult [the breaking changes page](../docs/breaking-changes).
+
+The good news is that it will be much more simple to use and much more flexible.
+And it produces pretty pictures.
+
+## Keys
+
+A [much][] [demanded][] [feature][] is the ability to define so-called bootvars.
+Bootvars are variables which value would be set either at configure time or at
+startup time.
+
+[much]: https://github.com/mirage/mirage/issues/229
+[demanded]: https://github.com/mirage/mirage/issues/228
+[feature]: https://github.com/mirage/mirage/issues/231
+
+
+A good example is the ip address of the http stack, you want to be able to:
+
+- Set a good default directly in the `config.ml`
+- Provide a value at configure time, if you are already aware of deployment conditions.
+- Provide a value at startup time, for last minute changes.
+
+All of this is now possible using **keys**. A key is composed of :
+- _name_ : The name of the value in the program.
+- _description_ : How it should be displayed/serialized.
+- _stage_ : Is the key available only at runtime, at configure time or both ?
+- _documentation_ : It is not optional so you should really write it.
+
+Imagine we are building a multilingual unikernel and we want to pass the
+default language as a parameter. We will use a simple string, so we can use the
+predefined description `Key.Desc.string`. We want to be able to define it both
+at configure and run time, so we use the stage `` `Both``. This gives us the following code:
+
+```
+let lang_key =
+  let doc = Key.Doc.create
+      ~doc:"The default language for the unikernel." [ "l" ; "lang" ]
+  in
+  Key.create ~doc ~stage:`Both ~default:"en" "language" Key.Desc.string
+```
+
+Here, We defined both a long option `--lang` and a short one `-l` (the format is similar to the one used by [Cmdliner][cmdliner]).
```

Comment:
`Here, We` -> `Here, we`.
(And I think you could just say `[Cmdliner][]` if you care.)